run/jobs/main_test.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2022 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+import subprocess
+
+import main
+
+
+def test_main(capsys):
+    main.main()
+    out, _ = capsys.readouterr()
+    expected = "Completed Task #0."
+    assert expected in out
+
+
+def test_env_vars():
+    with pytest.raises(Exception, match=r".*failed.*"):
+        main.main(fail_rate="0.999999")
+
+
+def test_bad_env_vars():
+    with pytest.raises(Exception, match=r".*Invalid.*"):
+        main.main(fail_rate="2")
+
+
+def test_run_script():
+    output = (
+        subprocess.run(
+            ["python", "main.py"],
```

Comment:
Not sure if you want to add a note here about python aliases?  If `python` is pointing at `2.*`, then it will fail because our samples only support python3.   Locally, I have to change this to `python3` to make it work.